auto_nag/scripts/topcrash_add_keyword.py

```diff
@@ -0,0 +1,165 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+import re
+
+from libmozdata.bugzilla import Bugzilla
+from libmozdata.connection import Connection
+
+from auto_nag import utils
+from auto_nag.bzcleaner import BzCleaner
+from auto_nag.topcrash import Topcrash
+
+# TODO: should be moved when resolving https://github.com/mozilla/relman-auto-nag/issues/1384
+HIGH_SEVERITY = {"S1", "critical", "S2", "major"}
+
+MAX_SIGNATURES_PER_QUERY = 30
+
+
+class TopcrashAddKeyword(BzCleaner):
+    def description(self):
+        return "Bugs with missing topcrash keywords"
+
+    def columns(self):
+        return ["id", "summary", "severity", "added_keyword"]
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+        if bugid in data:
+            return
+
+        has_topcrash_keyword = "topcrash" in bug["keywords"]
+        has_startup_keyword = "topcrash-startup" in bug["keywords"]
+        if has_topcrash_keyword and has_startup_keyword:
+            return
+
+        signatures = utils.get_signatures(bug["cf_crash_signature"])
+        has_topcrash_signature = any(
+            signature in self.topcrashes for signature in signatures
+        )
+        assert has_topcrash_signature
+
+        has_startup_signature = any(
+            self.topcrashes.get(signature) for signature in signatures
+        )
+
+        should_add_keyword = (
+            has_topcrash_signature
+            and not (has_topcrash_keyword or has_startup_signature)
+        ) or (has_startup_signature and not has_startup_keyword)
```

Comment:
This seems too complex and prone to error. The first condition (`has_topcrash_signature`) is unneeded because we already assert it is True a few lines earlier.

What if you explicity list all conditions in which we don't have to do anything, with explanatory comments?
E.g.
```Python
# We already have a topcrash keyword and the crash is not a startup crash.
if has_topcrash_keyword and not has_startup_signature:
    return

# bla bla bla
elif XXX
   return
```

Reply:
Don in fcc95573ad41136bf0be5c583ac7f7ae5d2dc59f

Reply:
Improved in 867d4afd1b1588717f0f79c5c6b7eb5f13243438

Reply:
Improved in 867d4afd1b1588717f0f79c5c6b7eb5f13243438